libraries/AP_Scripting/applets/param-set.lua

```diff
@@ -0,0 +1,165 @@
+
+-- initialize MAVLink rx with buffer depth and number of rx message IDs to register
+mavlink:init(5, 1)
+
+-- register message id to receive
+local PARAM_SET_ID = 23
+mavlink:register_rx_msgid(PARAM_SET_ID)
+
+-- handle PARAM_SET message
+local parameters_which_can_be_set = {}
+parameters_which_can_be_set["MAV_OPTIONS"] = true
+parameters_which_can_be_set["PARAM_SET_ENABLE"] = true
+parameters_which_can_be_set["BATT_ARM_MAH"] = true
+parameters_which_can_be_set["BATT_ARM_VOLT"] = true
+parameters_which_can_be_set["BATT_CAPACITY"] = true
+parameters_which_can_be_set["BATT_CRT_MAH"] = true
+parameters_which_can_be_set["BATT_CRT_VOLT"] = true
+parameters_which_can_be_set["BATT_FS_CRT_ACT"] = true
+parameters_which_can_be_set["BATT_FS_LOW_ACT"] = true
+parameters_which_can_be_set["BATT_LOW_MAH"] = true
+parameters_which_can_be_set["BATT_LOW_VOLT"] = true
+parameters_which_can_be_set["BRD_OPTIONS"] = true
+parameters_which_can_be_set["COMPASS_USE3"] = true
+parameters_which_can_be_set["FENCE_ACTION"] = true
+parameters_which_can_be_set["FENCE_ALT_MAX"] = true
+parameters_which_can_be_set["FENCE_ENABLE"] = true
+parameters_which_can_be_set["FENCE_RADIUS"] = true
+parameters_which_can_be_set["FENCE_TYPE"] = true
+parameters_which_can_be_set["LIGHTS_ON"] = true
+parameters_which_can_be_set["LOG_BITMASK"] = true
+parameters_which_can_be_set["LOG_DISARMED"] = true
+parameters_which_can_be_set["LOG_FILE_DSRMROT"] = true
+parameters_which_can_be_set["RTL_ALT"] = true
+parameters_which_can_be_set["RTL_LOIT_TIME"] = true
+parameters_which_can_be_set["RTL_SPEED"] = true
+
+local function should_set_parameter_id(param_id)
+    if parameters_which_can_be_set[param_id] == nil then
+        return false
+    end
+    return true
+end
+
+local function handle_param_set(name, value)
+    gcs:send_text(MAV_SEVERITY.WARNING, string.format("%s: param set received", TEXT_PREFIX_STR))
+
+    -- we will not receive packets in here for the wrong system ID /
+    --   component ID; this is handled by ArduPilot's MAVLink routing
+    --   code
+
+    -- check for this specific ID:
+    if not should_set_parameter_id(name) then
+        gcs:send_text(MAV_SEVERITY.WARNING, string.format("%s: param set denied (%s)", TEXT_PREFIX_STR, name))
+        return
+    end
+
+    param:set_and_save(name, value)
+end
+
+-- display welcome message
+gcs:send_text(MAV_SEVERITY.INFO, "param-set script loaded")
+
+-- initialise our knowledge of the GCS's allow-set-parameters state.
+--   We do not want to fight over setting this GCS state via other
+--   mechanisms (eg. an auxiliary function), so we keep this state
+--   around to track what we last set:
+local gcs_allow_set = gcs:get_allow_param_set()
+
+-- update function to receive param_set messages and perhaps act on them
+local function update()
+    -- return immediately if not enabled
+    if (PARAM_SET_ENABLE:get() <= 0) then
+        -- this script is disabled, set allow-via-GCS (once):
```

Comment:
Should probably still drain the buffer in this case. Otherwise when you enable the earlier sets will suddenly go through.

Reply:
Added a loop to drain the messages.